scripts/Dockerfile.cardinal-src-build

```diff
@@ -0,0 +1,66 @@
+# Build and install Cardinal in a Docker image without conda
+#
+# Divided into the following targets:
+#
+# cardinal-base: starting from an Ubuntu 24.04 image:
+#     * installs all required packages
+#
+# cardinal-clone:
+#     * clone cardinal repo
+#
+# cardinal-deps:
+#     * run the get-dependency script
+#
+# cardinal-build:
+#     * make cardinal using specific environment variables
+#
+FROM ubuntu:24.04 AS cardinal-base
+
+# Install necessary dependencies
+RUN apt-get update --yes && \
+    apt-get install --yes \
+        git \
+        make \
+        autoconf \
+        automake \
+        libtool \
+        flex \
+        bison \
+        cmake \
+        g++ \
+        gfortran \
+        libhdf5-dev \
+        mpich \
+        libtirpc-dev \
+        python3 \
+        python3-pip \
+        python3-packaging \
+        python3-jinja2 \
+        python3-yaml \
+        python3-pkgconfig
+
+FROM cardinal-base AS cardinal-clone
+
+# COPY . /cardinal    
+WORKDIR /cardinal-build
+RUN git clone https://github.com/neams-th-coe/cardinal.git
+
+FROM cardinal-clone AS cardinal-deps
+
+WORKDIR /cardinal-build/cardinal
+
+ENV LIBMESH_JOBS=16
```

Comment:
This doesn't seem to be accelerating the build

Reply:
I believe you also need to set `JOBS` and `MOOSE_JOBS` to accelerate a full build from source. From my understanding `update_and_rebuild_libmesh.sh` overrides `LIBMESH_JOBS` in a few instances with those environment variables when building libMesh from source.

Reply:
That definitely made a difference

Reply:
@nuclearkevin thanks for this clarification! I updated the documentation (#994) to make sure its comprehensive.